src/snowflake/connector/aio/auth/_idtoken.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from ...auth.idtoken import AuthByIdToken as AuthByIdTokenSync
+from ._by_plugin import AuthByPlugin
+from ._webbrowser import AuthByWebBrowser
+
+if TYPE_CHECKING:
+    from .._connection import SnowflakeConnection
+
+
+class AuthByIdToken(AuthByPlugin, AuthByIdTokenSync):
+    def __init__(
+        self,
+        id_token: str,
+        application: str,
+        protocol: str | None,
+        host: str | None,
+        port: str | None,
+        **kwargs,
+    ) -> None:
+        """Initialized an instance with an IdToken."""
+        AuthByIdTokenSync.__init__(
+            self, id_token, application, protocol, host, port, **kwargs
+        )
+
+    async def reset_secrets(self) -> None:
+        AuthByIdTokenSync.reset_secrets(self)
+
+    async def prepare(self, **kwargs: Any) -> None:
```

Comment:
do we need to make these two function async if its implementation does not contain any async code?